### DISCUSSION TITLE
robots.txt | Next.js | public - folder

### DISCUSSION THREAD

**Discussion by enkicoma** [vercel/next.js#19495] [Help] [answered]
Hi guys,

may I ask if someone can share a proper `robots.txt` to put it in my `public` folder, as I understood this is great to have for some bad `bots` on the internet...

I may rise a PR to include it for Three-example because there are none:
https://github.com/vercel/next.js/tree/canary/examples/with-three-js

Cheers!

**Comment by enkicoma** [comment]
> Hi guys,
> 
> may I ask if someone can share a proper `robots.txt` to put it in my `public` folder, as I understood this is great to have for some bad `bots` on the internet...
> 
> I may rise a PR to include it for Three-example because there are none:
> https://github.com/vercel/next.js/tree/canary/examples/with-three-js
> 
> Cheers!

nevermind, I understood that this `robots.txt` file is more personal, web-prod preference, to allow or to deny a specific path or file, etc
https://developers.google.com/search/docs/advanced/robots/intro?hl=en&visit_id=637418414571993647-2823432621&rd=1#what-is-a-robots.txt-file

**Comment by Tomburgs** [comment, marked as answer]
There's no such thing as a "proper" `robots.txt`, each website has different requirements.

It's just a hint to the crawlers what they should and should not be looking at, for example if I was running an E-Commerce store, I would want to tell the crawlers to not crawl the `/checkout` page.
Similarly, I could generate a sitemap telling the crawlers all the pages I would like them to crawl.

Do note, however, that the robots.txt does not have any real & enforceable power, it's merely a text file that the crawlers choose to respect.

[This website](http://www.robotstxt.org) might be a useful reference for you when creating a `robots.txt` file :)

**Reply by enkicoma** [reply]
Cheers Tom!

**Reply by enkicoma** [reply]
I took an example from apple:
https://www.apple.com/robots.txt

by any mean do you know what is: `DaumWebMasterTool:d7****` 

can't find on google

**Reply by Tomburgs** [reply]
It appears that there's a Korean search engine named [Daum](https://daum.net).
It must be a verification token, so that Apple's SEO team could verify their ownership of this domain & get insights on their rankings on this website.

There's similar things you can do for Google & Bing, but those typically include specifying a TXT record on your name servers, doing it in the robotstxt is the first I've seen.

**Reply by enkicoma** [reply]
interesting and make sense...
Once again, thanks!
